docs/software_requirements/system_initialization.sdoc

```diff
@@ -10,3 +10,269 @@ IMPORT_FROM_FILE: software_requirements.sgra
 TBD
 
 [/FREETEXT]
+
+[SECTION]
+TITLE: Interface Requirements
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[REQUIREMENT]
+UID: ZEP-127
+STATUS: Draft
+TYPE: Interface
+COMPONENT: System Initialization
+TITLE: Build-time Registration for System Initialization Notifications
+STATEMENT: >>>
+The Zephyr RTOS shall expose an interface to build-time registration for the following system initialization notifications:
+
+- Pre-CPU Architecture Initialization
+- Pre-Kernel Initialization
+- Post-Kernel Initialized
+- Application Initialization
+<<<
```

Comment:
There needs to be a symmetric set of events for system shutdown.

The shutdown event registration is needed to enable a bootloader to register a handler for handing off execution to the loaded image while still allowing a controlled, clean shutdown of the RTOS.